compute/cloud-client/src/main/java/compute/custommachinetype/CreateWithHelper.java

```diff
@@ -0,0 +1,307 @@
+          .setProject(project)
+          .setZone(zone)
+          .setInstanceResource(instance)
+          .build();
+
+      // Invoke the API with the request object and wait for the operation to complete.
+      Operation response = instancesClient.insertAsync(insertInstanceRequest)
+          .get(3, TimeUnit.MINUTES);
+
+      // Check for errors.
+      if (response.hasError()) {
+        throw new Error("Instance creation failed!!" + response);
+      }
+      System.out.printf("Instance created : %s", instanceName);
+      System.out.println("Operation Status: " + response.getStatus());
+    }
+  }
+
+  static int gbToMb(int value) {
+    return value << 10;
+  }
+
+  static int[] concat(int[] a, int[] b) {
+    int[] result = new int[a.length + b.length];
+    System.arraycopy(a, 0, result, 0, a.length);
+    System.arraycopy(b, 0, result, a.length, b.length);
+    return result;
+  }
+
+  public enum CpuSeries {
+    N1("custom"),
+    N2("n2-custom"),
+    N2D("n2d-custom"),
+    E2("e2-custom"),
+    E2_MICRO("e2-custom-micro"),
+    E2_SMALL("e2-custom-small"),
+    E2_MEDIUM("e2-custom-medium");
+
+    private static final Map<String, CpuSeries> ENUM_MAP;
+
+    static {
+      ENUM_MAP = init();
+    }
+
+    // Build an immutable map of String name to enum pairs.
+    public static Map<String, CpuSeries> init() {
+      Map<String, CpuSeries> map = new ConcurrentHashMap<>();
+      for (CpuSeries instance : CpuSeries.values()) {
+        map.put(instance.getCpuSeries(), instance);
+      }
+      return Collections.unmodifiableMap(map);
+    }
+
+    private final String cpuSeries;
+
+    CpuSeries(String cpuSeries) {
+      this.cpuSeries = cpuSeries;
+    }
+
+    public static CpuSeries get(String name) {
+      return ENUM_MAP.get(name);
+    }
+
+    public String getCpuSeries() {
+      return this.cpuSeries;
+    }
+  }
+
+  // This enum correlates a machine type with its limits.
+  // The limits for various CPU types are described in:
+  // https://cloud.google.com/compute/docs/general-purpose-machines
+  enum Limits {
+    CPUSeries_E2(new TypeLimits(getNumsInRangeWithStep(2, 33, 2), 512, 8192, false, 0)),
+    CPUSeries_E2MICRO(new TypeLimits(new int[]{}, 1024, 2048, false, 0)),
+    CPUSeries_E2SMALL(new TypeLimits(new int[]{}, 2048, 4096, false, 0)),
+    CPUSeries_E2MEDIUM(new TypeLimits(new int[]{}, 4096, 8192, false, 0)),
+    CPUSeries_N2(
+        new TypeLimits(concat(getNumsInRangeWithStep(2, 33, 2), getNumsInRangeWithStep(36, 129, 4)),
+            512, 8192, true, gbToMb(624))),
+    CPUSeries_N2D(
+        new TypeLimits(new int[]{2, 4, 8, 16, 32, 48, 64, 80, 96}, 512, 8192, true, gbToMb(768))),
+    CPUSeries_N1(
+        new TypeLimits(concat(new int[]{1}, getNumsInRangeWithStep(2, 97, 2)), 922, 6656, true,
+            gbToMb(624)));
+
+    private final TypeLimits typeLimits;
+
+    Limits(TypeLimits typeLimits) {
+      this.typeLimits = typeLimits;
+    }
+
+    public TypeLimits getTypeLimits() {
```

Comment:
Do we have a precedence as to what order to place the different parts of the code? I see
- Variable initializations
- Static block initialization
- Enums
-  Inner classes
- Methods
- Main 

spread across the class. was there any specific guidance followed? If not can we structure it as:

- inner class
- enums
- class variables
- static block initialization
- main
- methods

Reply:
This comment applies to multiple files

Reply:
Thanks for bringing this up. This is how it was supposed to be structured. But I guess java lint formatter mixed this up for me! 

Modified the format in the latest commit. PTAL.